video/transmux.go

```diff
@@ -20,75 +24,86 @@ func MuxTStoMP4(tsInputFile, mp4OutputFile string) ([]string, error) {
 		}).
 		OverWriteOutput().ErrorToStdOut().Run()
 	if err != nil {
-		return nil, fmt.Errorf("failed to transmux concatenated mpeg-ts file (%s) into a mp4 file: %s", tsInputFile, err)
+		return nil, fmt.Errorf("failed to transmux concatenated mpeg-ts file (%s) into a mp4 file: %w", tsInputFile, err)
 	}
 	// Verify the mp4 output file was created
 	_, err = os.Stat(mp4OutputFile)
 	if err != nil {
-		return nil, fmt.Errorf("transmux error: failed to stat MP4 media file: %s", err)
+		return nil, fmt.Errorf("transmux error: failed to stat MP4 media file: %w", err)
 	} else {
 		transmuxOutputFiles = append(transmuxOutputFiles, mp4OutputFile)
 	}
 	return transmuxOutputFiles, nil
 }
 
-func MuxTStoFMP4WithHLS(tsInputFile, fmp4ManifestOutputFile string) ([]string, error) {
-
+func MuxTStoFMP4(fmp4ManifestOutputFile string, inputs ...string) error {
 	baseFragMp4Dir := filepath.Dir(fmp4ManifestOutputFile)
 	err := os.Mkdir(baseFragMp4Dir, 0700)
 	if err != nil && !os.IsExist(err) {
-		return nil, fmt.Errorf("transmux error: failed to create subdir to output fmp4 files: %s", err)
+		return fmt.Errorf("transmux error: failed to create subdir to output fmp4 files: %w", err)
 	}
 
-	var transmuxOutputFiles []string
-	// transmux the .ts file into fMP4 packaged with HLS
-	err = ffmpeg.Input(tsInputFile).
-		Output(fmp4ManifestOutputFile, ffmpeg.KwArgs{"movflags": "frag_keyframe+empty_moov",
-			"c":                 "copy",
-			"bsf:a":             "aac_adtstoasc",
-			"hls_time":          10,
-			"hls_playlist_type": "vod",
-			"hls_segment_type":  "fmp4",
-			"hls_flags":         "single_file"}).
-		OverWriteOutput().ErrorToStdOut().Run()
-	if err != nil {
-		return nil, fmt.Errorf("transmux error: failed to transmux concatenated mpeg-ts file (%s) into a fragemented-mp4 file: %s", tsInputFile, err)
+	var args []string
+	mapArgs := []string{"-map", "0:a"}
+	for i, input := range inputs {
+		args = append(args, "-i", input)
+		mapArgs = append(mapArgs, "-map", fmt.Sprintf("%d:v", i))
 	}
+	args = append(args,
+		"-movflags", "frag_keyframe+empty_moov",
+		"-c", "copy",
+		"-bsf:a", "aac_adtstoasc",
+		"-f", "dash",
+		"-dash_segment_type", "mp4",
+		"-single_file", "1",
+		"-hls_playlist", "1",
+		"-hls_time", "10",
+		"-hls_playlist_type", "vod",
+		"-hls_segment_type", "fmp4",
+		"-vtag", "avc1",
+		"-atag", "mp4a",
```

Comment:
Would like to figure out how to avoid needing these two params. Ffmpeg errors without them (when using ts input rather than mp4)